src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -258,6 +259,60 @@ def __len__(self) -> int:
         """
         return len(self._buffer)
 
+    def window(
+        self,
+        start: int | datetime,
+        end: int | datetime,
+        fill_gaps: Optional[float] = None,
+        force_copy: bool = False,
+    ) -> ArrayLike:
+        """
+        Access a sub window of the `MovingWindow`.
+
+        Args:
+            start: The start index or datetime of the sub window.
+            end: The end index or datetime of the sub window.
+            fill_gaps: The value to fill gaps with.
+            force_copy: If True, a copy of the underlying buffer is returned.
+
+        Raises:
+            IndexError: if the end index is smaller than the start index.
+
+        Returns:
+            A sub window of the `MovingWindow`.
+        """
+        if len(self._buffer) == 0:
+            raise IndexError("The buffer is empty.")
+        if isinstance(start, datetime) and isinstance(end, datetime):
+            if start > end:
+                raise IndexError(
+                    f"end parameter {end} has to predate start parameter {start}"
+                )
+
+            start_index = self._buffer.datetime_to_index(start)
+            end_index = self._buffer.datetime_to_index(end)
+        else:
+            start_index = start  # type: ignore
+            end_index = end  # type: ignore
+
+        # Make copy of buffer
+        buffer_to_use = deepcopy(self._buffer) if (force_copy) else self._buffer
+
+        # Fill gaps with the given value
+        # Assumption that if fill_gaps is used then also force_copy is used
+        # To make this even more efficient use interval tree
+        if fill_gaps is not None and force_copy:
+            for gap_range in buffer_to_use.gaps:
+                buffer_to_use[gap_range.start : gap_range.end] = [fill_gaps] * len(range(gap_range.start, gap_range.end))  # type: ignore  # pylint: disable=line-too-long
```

Comment:
I'm not sure if we should disable the line-too-long rule. @llucax what do you think?

Reply:
I don't see a reason to remove it, `black` should fix most cases of lines being too long, and it has a soft limit of 88 but I think it allows going up to 110 chars, which IMHO is reasonable. What might happen is that `pylint` is not properly configured and complaining for lines longer than 88.